src/main/java/org/symphonyoss/symphony/messageml/elements/Password.java

```diff
@@ -0,0 +1,154 @@
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class representing a Password Field inside a Form.
+ * @author Sandro Ribeiro
+ * @since 06/13/2019
+ */
+public class Password extends FormElement {
+
+  public static final String MESSAGEML_TAG = "password";
+
+  private static final String MINLENGTH_ATTR = "minlength";
+  private static final String MAXLENGTH_ATTR = "maxlength";
+  private static final String REQUIRED_ATTR = "required";
+  private static final String PLACEHOLDER_ATTR = "placeholder";
+
+  private static final List<String> VALID_VALUES_FOR_REQUIRED_ATTR = Arrays.asList("true", "false");
+
+  private final static String MARKDOWN = "Password Field";
+
+  private static final String PRESENTATIONML_INPUT_TAG = "input";
+  private static final String PRESENTATIONML_TYPE_ATTR = "type";
+  private static final String PRESENTATIONML_INPUT_TYPE = "password";
+
+  private static final int MIN_ALLOWED_LENGTH = 1;
+  private static final int MAX_ALLOWED_LENGTH = 128;
+
+  public Password(Element parent) {
+    super(parent, MESSAGEML_TAG);
+  }
+
+  @Override
+  public void validate() throws InvalidInputException {
+    super.validate();
+
+    if (getAttribute(NAME_ATTR) == null) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
+    if (getAttribute(REQUIRED_ATTR) != null) {
+      assertAttributeValue(REQUIRED_ATTR, VALID_VALUES_FOR_REQUIRED_ATTR);
+    }
+
+    validateMinAndMaxLengths();
+    assertNoContent();
+  }
+
+  @Override
+  protected void buildAttribute(org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+      case NAME_ATTR:
+        setAttribute(NAME_ATTR, getStringAttribute(item));
+        break;
+      case REQUIRED_ATTR:
+        setAttribute(REQUIRED_ATTR, getStringAttribute(item));
+        break;
+      case PLACEHOLDER_ATTR:
+        setAttribute(PLACEHOLDER_ATTR, getStringAttribute(item));
+        break;
+      case MINLENGTH_ATTR:
+        setAttribute(MINLENGTH_ATTR, getStringAttribute(item));
+        break;
+      case MAXLENGTH_ATTR:
+        setAttribute(MAXLENGTH_ATTR, getStringAttribute(item));
+        break;
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    Map<String, String> presentationAttrs = buildTextFieldInputAttributes();
+    out.printElement(PRESENTATIONML_INPUT_TAG, presentationAttrs);
+  }
+
+  @Override
+  public org.commonmark.node.Node asMarkdown() {
+    if (getAttribute(PLACEHOLDER_ATTR) != null) {
+      return new FormElementNode(MARKDOWN, ":" + getAttribute(PLACEHOLDER_ATTR));
+    } else {
+      return new FormElementNode(MARKDOWN, "");
```

Comment:
You don't need to pass the empty string anymore.

Reply:
Ouch, I forgot about it...